src/org/zaproxy/zap/extension/advreport/AlertsPanel.java

```diff
@@ -36,51 +44,98 @@ private void initialize( List<String> alertTypes ){
 		gbc.gridx = 0;
 		gbc.gridy = 0;
 		gbc.fill = GridBagConstraints.HORIZONTAL;
-		
+		gbc.insets = new Insets(2,5,30,10);
+
+		selectionPanel.add(getRiskBox("High"), gbc);
+
+		gbc.gridx++;
+		selectionPanel.add(getRiskBox("Medium"), gbc);
+
+		gbc.gridx++;
+		selectionPanel.add(getRiskBox("Low"), gbc);
+
+		gbc.gridx++;
+		selectionPanel.add(getRiskBox("Informational"), gbc);
+
+		gbc.insets = new Insets(0,20,2,0);
+		gbc.gridy = 1;
+		gbc.gridwidth = 4;
+
 		for( String alertType: alertTypes ){
-			JLabel label = new JLabel(alertType);
-		    JCheckBox selection = new JCheckBox();
-		    selection.setSelected( true );
-		    selection.setName( alertType );
-		    
-		    gbc.gridx = 0;
-		    gbc.anchor = GridBagConstraints.WEST;
-		    selectionPanel.add(label, gbc);
-			
-		    gbc.gridx = 1;
-		    gbc.anchor = GridBagConstraints.EAST;
-		    selectionPanel.add(selection, gbc);
-		    this.getSelections().add( selection );	
-			
+
+			JCheckBox selection = new JCheckBox();
+			selection.setText(alertType);
+			selection.setSelected( true );
+			selection.setName( alertType );
+
+			gbc.gridx = 0;
+			gbc.anchor = GridBagConstraints.CENTER;
+			selectionPanel.add(selection, gbc);
+			this.getSelections().add( selection );
 			gbc.gridy += 1;
+
 		}
-		 				
+
         JPanel buttonpane = new JPanel( new FlowLayout( FlowLayout.RIGHT ));
         buttonpane.add( getCancelButton() );
         buttonpane.add( getHTMLButton() );
-        
+
 		this.setLayout( new BorderLayout() );
 		this.add( new JLabel(" Alerts in Report : " ), BorderLayout.NORTH );
 		this.add( new JScrollPane( selectionPanel ), BorderLayout.CENTER );
 	    this.add(buttonpane, BorderLayout.SOUTH);
-		
+
 	}
-	
+
 	private List<JCheckBox> getSelections(){
 		if( selections == null ){
 			selections = new ArrayList<JCheckBox>();
 		}
 		return selections;
 	}
-	
+
 	public List<String> getSelectedAlerts(){
 		List<String> selectedAlerts = new ArrayList<String>();
 		for( JCheckBox selection: selections ){
 			if( selection.isSelected() ) selectedAlerts.add( selection.getName());
 		}
 		return selectedAlerts;
+
+	}
+
+	private JCheckBox getRiskBox(final String riskLevel){
+		JCheckBox riskChk = new JCheckBox();
+		riskChk.setText(riskLevel);
+		riskChk.setSelected(true);
+		riskChk.addItemListener(new java.awt.event.ItemListener() {
+			@Override
+			public void itemStateChanged(java.awt.event.ItemEvent e) {
+				Boolean selected = (ItemEvent.SELECTED == e.getStateChange());
+				List<String> alertTypes = getRiskAlertTypes(riskLevel);
+				for (JCheckBox selection: selections ){
+					if(alertTypes.contains(selection.getName())) {
+						selection.setSelected(selected);
+					}
+				}
+			}
+		});
+
+		return riskChk;
 	}
-	
+
+    private List<String> getRiskAlertTypes(String riskLevel) {
+    	ExtensionAlert extAlert = (ExtensionAlert) Control.getSingleton().getExtensionLoader().getExtension(ExtensionAlert.NAME);
+        List<Alert> alerts = extAlert.getAllAlerts();

```

Comment:
This will read all alerts into memory which might easily lead to OOME. Since the ```ExtensionAlert``` does not (yet) allow pagination, you should do it yourself by reading the IDs of the alerts and build the ```Alert```s as needed (take a look at the implementation of ```ExtensionAlert.getAllAlerts()``` on how to do that).
Also, reads from database should be minimised, doing it each time the user changes the selected risks is overkill. It's better to cache the name/risk of the alerts and just use that.
Since all alerts were already read in ```ExtensionAdvReport.getAlertTypes()``` it's better to cache there and pass the cached data to this class.